tests/integration/otel/test_otel.py

```diff
@@ -0,0 +1,1234 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Such thorough tests are greatly appreciated.

Reply:
Thank you! These changes are so complicated that require such integration tests to make sure that everything works as expected.